src/com/trilead/ssh2/transport/TransportManager.java

```diff
@@ -688,7 +682,7 @@ public void receiveLoop() throws IOException
 	{
 		byte[] msg = new byte[MAX_PACKET_SIZE];
 
-		while (true)
+		while (!isConnectionClosed())
```

Comment:
Looking at the code, this acquires a lock every time -- wondering if there is a more efficient way of doing this check?